Document `DD_PROFILING_MAX_FRAMES` / `max_frames` setting for Ruby profiler (#16361)
DataDog/documentation: content/en/profiler/profiler_troubleshooting.md

There is a known incompatibility between Ruby 2.7 and older GCC versions (4.8 and below) that impacts the profiler ([upstream Ruby report][6], [`dd-trace-rb` bug report][7]). This can result in the following error message: "Your ddtrace installation is missing support for the Continuous Profiler because compilation of the Ruby VM just-in-time header failed. Your C compiler or Ruby VM just-in-time compiler seem to be broken."

To fix this, update your operating system or Docker image so that the GCC version is something more recent than v4.8.

For further help with this issue, [contact support][2] and include the output of running `DD_PROFILING_FAIL_INSTALL_IF_MISSING_EXTENSION=true gem install ddtrace` and the resulting `mkmf.log` file.

## Frames omitted when backtraces are very deep

> The Ruby profiler truncates deep backtraces when collecting profiling data. Truncated backtraces are missing some of their caller functions making it impossible to link them to the root call frame. As a result, truncated backtraces are grouped together under a "N frames omitted" frame.

Reviewer: ```suggestion
The Ruby profiler truncates deep backtraces when collecting profiling data. Truncated backtraces are missing some of their caller functions, making it impossible to link them to the root call frame. As a result, truncated backtraces are grouped together under a `N frames omitted` frame.
```

> The maximum depth can be increased using the `DD_PROFILING_MAX_FRAMES` environment variable, or via code:

Reviewer: ```suggestion
You can increase the maximum depth with the `DD_PROFILING_MAX_FRAMES` environment variable, or in code:
```
